ui/src/data_explorer/components/WriteDataForm.js

```diff
@@ -0,0 +1,112 @@
+import React, {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import DatabaseDropdown from 'shared/components/DatabaseDropdown'
+
+import {writeData} from 'src/data_explorer/apis'
+import {publishAutoDismissingNotification} from 'shared/dispatchers'
+import {errorThrown as errorThrownAction} from 'shared/actions/errors'
+
+class WriteDataForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedDatabase: null,
+    }
+
+    this.handleSelectDatabase = ::this.handleSelectDatabase
+    this.handleError = ::this.handleError
+    this.handleWrite = ::this.handleWrite
+  }
+
+  handleSelectDatabase(item) {
+    this.setState({selectedDatabase: item.text})
+  }
+
+  handleError(error) {
+    const {errorThrown} = this.props
+    errorThrown(error)
+  }
+
+  async handleWrite() {
+    const {onClose, source, notify, errorThrown} = this.props
+    const {selectedDatabase} = this.state
+    try {
+      await writeData(source, selectedDatabase, this.editor.value)
+      notify('success', 'Data was written successfully')
+      onClose()
+    } catch (response) {
+      errorThrown(response, response.data.error)
+    }
+  }
+
+  render() {
+    const {onClose} = this.props
+    const {selectedDatabase} = this.state
+
+    return (
+      <div className="write-data-form">
+        <div className="write-data-form--header">
+          <div className="page-header__left">
+            <h1 className="page-header__title">Write Data To</h1>
+            <DatabaseDropdown
+              onSelectDatabase={this.handleSelectDatabase}
+              database={selectedDatabase}
+              onStartEdit={() => {}}
```

Comment:
I'm going to make it optional in `DatabaseDropdown` and remove it from here, it's a required prop but doesn't look like it's necessary, it's just being passed to `Dropdown`, which doesn't require it.